clang/runtime/dpct-rt/processFiles.py

```diff
@@ -0,0 +1,127 @@
+        inc_files_all_dir_result = os.path.join(inc_files_dir, "dpl_extras/", runtime_src_file + ".inc")
+        return [cont_file_result, inc_files_all_dir_result]
+    else:
+        cont_file_result = os.path.join(runtime_header_src_files_dir, runtime_src_file)
+        inc_files_all_dir_result = os.path.join(inc_files_dir, runtime_src_file + ".inc")
+        return [cont_file_result, inc_files_all_dir_result]
+
+
+def create_dir(inc_files_dir):
+    if (not os.path.exists(os.path.join(inc_files_dir))):
+        os.makedirs(os.path.join(inc_files_dir))
+    if (not os.path.exists(os.path.join(inc_files_dir, "dpl_extras/"))):
+        os.makedirs(os.path.join(inc_files_dir, "dpl_extras/"))
+
+def convert_to_cxx_code(line_code):
+    return bytes("R\"Delimiter(", 'utf-8') + line_code + bytes(")Delimiter\"\n", 'utf-8')
+
+def process_a_file(runtime_src_file, inc_files_dir, is_dpl_extras):
+    file_names = get_file_paths(runtime_src_file, inc_files_dir, is_dpl_extras)
+    cont_file_handle = io.open(file_names[0], "rb")
+    inc_all_file_handle = io.open(file_names[1], "w+b")
+
+    inc_all_file_lines = []
+
+    for line in cont_file_handle:
+        inc_all_file_lines.append(convert_to_cxx_code(line))
+
+    # cl.exe will emit error if a literal string length >= 65535
+    # so we need convert the generated code from style
+    # {code}
+    # std::string code = "a very very long string";
+    # {code}
+    # to style
+    # {code}
+    # std::string code = std::string("a long string") + ... + std::string("a long string");
+    # {code}
+    # The simplest way is treat one line a std::string instance and then use
+    # "+" to connect them, but if the number of std::string instances is too large (like 7000),
+    # clang will crash.
+    # So we should generate code like
+    # {code}
+    # std::string code = std::string("as long as possible but < 65535") + ... + std::string("as long as possible but < 65535");
+    # {code}
+    string_literal_length_limit = 65535
+    string_literal_length_counter = 0
+    inc_all_file_str = bytes("std::string(", 'utf-8')
+    for line in inc_all_file_lines:
+        if (string_literal_length_counter + len(line) < string_literal_length_limit):
+            inc_all_file_str = inc_all_file_str + line
+            string_literal_length_counter = string_literal_length_counter + len(line)
+        else:
+            inc_all_file_str = inc_all_file_str + bytes(")+std::string(", 'utf-8')
+            inc_all_file_str = inc_all_file_str + line
+            string_literal_length_counter = len(line)
+    inc_all_file_str = inc_all_file_str + bytes(")", 'utf-8')
+
+    inc_all_file_handle.write(inc_all_file_str)
+
+    cont_file_handle.close()
+    inc_all_file_handle.close()
+
+def get_files(path):
+    items = os.listdir(path)
+    files = []
+    for item in items:
+        if (os.path.isfile(os.path.join(path, item))):
+            files.append(item)
+    return files
+
+def main():
```

Comment:
Please further improve the python scripts on readme to include desc and howto.

Reply:
OK, I will update it in a separate PR.